### PR TITLE
upgrade to django 2.0 and add mock urls/views for all the issues

### PR DESCRIPTION
**Pull request by lpmi-13**
this may be an incredibly uncouth method of upgrading to django 2.0, but figured I'd give it a bash and see what people think. Happy to change any of this and/or just make incremental changes on smaller feature branches to the original repo, but figured as we're starting more or less from square 1, might as well just start a new project 2.0-style.

Fully intending to refactor as we go.

### CODE REVIEW

File: oxfordpython/settings.py [hunk 1]
```diff
@@ -17,7 +17,13 @@
 
 
 # Quick-start development settings - unsuitable for production
-# See https://docs.djangoproject.com/en/1.9/howto/deployment/checklist/
+# See https://docs.djangoproject.com/en/2.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'y54h%=s#!+=#-oahz$spl9lhu(*c0y7@+^*c0)q+9g=7)5-9=8'
```
Comment (LilyFirefly) [inline oxfordpython/settings.py:23]: We should configure this with an environment variable. https://www.mattlayman.com/2017/12-factor-django-apps.html

File: oxfordpython/settings.py [hunk 2]
```diff
@@ -17,7 +17,13 @@
...
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (LilyFirefly) [inline oxfordpython/settings.py:26]: This can also be an environment variable.

**Review by LilyFirefly — CHANGES_REQUESTED** [review]

**Comment by craigloftus** [comment]
I'd stick to atomic PRs. It is much easier to review, merge and make progress.

From a quick scan for Django 2 support we only need the change to `MIDDLEWARE` and removing the reference to `SessionAuthenticationMiddleware`

**Comment by lpmi-13** [comment]
Thanks for the comments, Craig and Ian...just the kind of feedback I was looking for!

I'll get on this and hopefully have a revised PR posthaste.

**Comment by lpmi-13** [comment]
alright, made another feature branch for just the django version upgrade, and will close this and open a PR from there.

**Pull request closed without merging**